frontend/packages/coingecko/src/api.ts

```diff
@@ -0,0 +1,43 @@
+import type { AxiosResponse } from "axios";
+import axios from "axios";
+import { pipe } from "fp-ts/function";
+import { either, option } from "fp-ts";
```

Comment:
I'm loving the FP, shame that the final code isn't as nicely readable as in Rust or Haskell (not due to you, just due to javascript)

Reply:
It was a bit rushed, we could use a bit of point-free and leverage composition to make it better though. 

Will keep it in mind for future ones.